services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/update/UpdateExpressionUtils.java

```diff
@@ -136,9 +139,18 @@ private static Function<String, String> behaviorBasedValue(UpdateBehavior update
     /**
      * Simple utility method that can create an ExpressionNames map based on a list of attribute names.
      */
-    private static Map<String, String> expressionNamesFor(String... attributeNames) {
-        return Arrays.stream(attributeNames)
-                     .collect(Collectors.toMap(EnhancedClientUtils::keyRef, Function.identity()));
-    }
+    private static Map<String, String> expressionNamesFor(String attributeNames) {
```

Comment:
Minor comment : following the pattern of older code , what if we use below ?
```java
    private static Map<String, String> expressionNamesFor(String attributeNames) {
        if (attributeNames.contains(NESTED_OBJECT_UPDATE)) {
            return Arrays.stream(PATTERN.split(attributeNames))
                         .collect(Collectors.collectingAndThen(
                             Collectors.toMap(EnhancedClientUtils::keyRef, Function.identity()),
                             Collections::unmodifiableMap));
        } else {
            return Collections.singletonMap(keyRef(attributeNames), attributeNames);
        }
    }
```

Reply:
i can modify this

Reply:
modified